python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py

```diff
@@ -716,3 +717,14 @@ def _router_bgp_vpws(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
                     route_targets=EosCliConfigGen.RouterBgp.VpwsItem.RouteTargets(import_export=rt),
                     pseudowires=pseudowires,
                 )
+
+    def _router_bgp_aggregate_addresses(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
+        """Set structured config for bgp aggregate addresses."""
+        for tenant in self.shared_utils.filtered_tenants:
+            for vrf in tenant.vrfs:
```

Comment:
Instead of running over these seperately, we should rather parse this inside the VRFs. You can have this helper function if you prefer, but have the vrf as argument and call it from _router_bgp_vrfs()

Reply:
make sure to update a test where we build VRF default and compare with the CLI that we render the aggregate at the correct place (under VRF default, under address family ipv4 or just globally)